### ISSUE TITLE
[2.1.1 CE] Rollback/Restore deletes database (--db) backup file in ${webroot}/var/backups.

### ISSUE THREAD

**Issue by azeemism** [magento/magento2#6460]
This is a follow up on a separate issue that may be related to #6459 

Running `php bin/magento setup:rollback -c 1472780740_filesystem_code.tgz -m 1472780740_filesystem_media.tgz -d 1472780740_db.sql` as `www-data` in developer mode looks to correctly Rollback/Restore Magento, however the *_db.sql file created during the backup is deleted.
### Preconditions:
1. Magento 2.1.1 CE (Composer install)
2. php bin/magento deploy:mode:set developer
3. Mariadb

```
root@skynet:/# dpkg -l | grep mariadb
ii  libmariadbclient18                   10.1.16+maria-1~jessie               amd64        MariaDB database client library
ii  mariadb-client                       10.1.16+maria-1~jessie               all          MariaDB database client (metapackage depending on the latest version)
ii  mariadb-client-10.1                  10.1.16+maria-1~jessie               amd64        MariaDB database client binaries
ii  mariadb-client-core-10.1             10.1.16+maria-1~jessie               amd64        MariaDB database core client binaries
ii  mariadb-common                       10.1.16+maria-1~jessie               all          MariaDB database common files (e.g. /etc/mysql/conf.d/mariadb.cnf)
ii  mariadb-server                       10.1.16+maria-1~jessie               all          MariaDB database server (metapackage depending on the latest version)
ii  mariadb-server-10.1                  10.1.16+maria-1~jessie               amd64        MariaDB database server binaries
ii  mariadb-server-core-10.1             10.1.16+maria-1~jessie               amd64        MariaDB database core server files
```
1. PHP 7.0

```
root@skynet:/# dpkg -l | grep php7
ii  libapache2-mod-php7.0                7.0.9-1~dotdeb+8.1                   amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  php7.0                               7.0.9-1~dotdeb+8.1                   all          server-side, HTML-embedded scripting language (metapackage)
ii  php7.0-cli                           7.0.9-1~dotdeb+8.1                   amd64        command-line interpreter for the PHP scripting language
ii  php7.0-common                        7.0.9-1~dotdeb+8.1                   amd64        documentation, examples and common module for PHP
ii  php7.0-curl                          7.0.9-1~dotdeb+8.1                   amd64        CURL module for PHP
ii  php7.0-fpm                           7.0.9-1~dotdeb+8.1                   amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)
ii  php7.0-gd                            7.0.9-1~dotdeb+8.1                   amd64        GD module for PHP
ii  php7.0-igbinary                      1.2.1-1~2b7c703-1~dotdeb+8.2         amd64        igbinary serializer for PHP
ii  php7.0-imagick                       3.4.2-1~dotdeb+8.2                   amd64        Provides a wrapper to the ImageMagick library
ii  php7.0-intl                          7.0.9-1~dotdeb+8.1                   amd64        Internationalisation module for PHP
ii  php7.0-json                          7.0.9-1~dotdeb+8.1                   amd64        JSON module for PHP
ii  php7.0-mbstring                      7.0.9-1~dotdeb+8.1                   amd64        MBSTRING module for PHP
ii  php7.0-mcrypt                        7.0.9-1~dotdeb+8.1                   amd64        libmcrypt module for PHP
ii  php7.0-mysql                         7.0.9-1~dotdeb+8.1                   amd64        MySQL module for PHP
ii  php7.0-opcache                       7.0.9-1~dotdeb+8.1                   amd64        Zend OpCache module for PHP
ii  php7.0-pspell                        7.0.9-1~dotdeb+8.1                   amd64        pspell module for PHP
ii  php7.0-readline                      7.0.9-1~dotdeb+8.1                   amd64        readline module for PHP
ii  php7.0-redis                         3.0.0~rc1-1~dotdeb+8.2               amd64        PHP extension for interfacing with Redis
ii  php7.0-soap                          7.0.9-1~dotdeb+8.1                   amd64        SOAP module for PHP
ii  php7.0-xml                           7.0.9-1~dotdeb+8.1                   amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP
ii  php7.0-xsl                           7.0.9-1~dotdeb+8.1                   all          XSL module for PHP (dummy)
ii  php7.0-zip                           7.0.9-1~dotdeb+8.1                   amd64        Zip module for PHP
```
1. Debian

```
root@skynet:/# cat /proc/version
Linux version 3.16.0-4-amd64 (debian-kernel@lists.debian.org) (gcc version 4.8.4 (Debian 4.8.4-1) ) #1 SMP Debian 3.16.7-ckt25-2+deb8u3 (2016-07-02)
```

```
root@skynet:/# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 8.5 (jessie)
Release:        8.5
Codename:       jessie
```
1. ulimit

```
www-data@skynet:~$ ulimit -n
500000
www-data@skynet:~$ ulimit -Sn
500000
www-data@skynet:~$ ulimit -Hn
500000
```
### Steps to reproduce
1. Install magento 2.1.0 with composer
2. Upgrade to 2.1.1 (while in developer mode)

```
    "require": {
        "magento/product-community-edition": "2.1.1",
        "composer/composer": "@beta"
    },
```

```
    "minimum-stability": "stable",
    "prefer-stable": true,
    "repositories": [
        {
            "type": "composer",
            "url": "https://repo.magento.com/"
        }
```

```
composer update
php bin/magento setup:upgrade
php bin/magento setup:di:compile
php bin/magento --ansi setup:static-content:deploy en_CA
```

```
www-data@skynet:~/example.com/magento/2.x/2.x$ php bin/magento setup:db:status
All modules are up to date.
```

```
www-data@skynet:~/example.com/magento/2.x/2.x$ php bin/magento -V
Magento CLI version 2.1.1
```
1. Switch to or stay in developer mode `php bin/magento deploy:mode:set developer`
2. Create a backup `php bin/magento setup:backup --code --media --db` as `www-data`
3. Restore a backup with `php bin/magento setup:rollback -c 1472780740_filesystem_code.tgz -m 1472780740_filesystem_media.tgz -d 1472780740_db.sql` as `www-data`

NOTE: Steps 1 (install 2.1.0 CE) & 2 (upgrade) may not be required to reproduce the issue, and I also expect the same issue if a backup is run in production mode (but I have not tested with production mode).
### Expected results
1. Database file `*_db.sql` in `${webroot}/var/backups` would remain in the backup location for a future Rollback/Restore
### Actual results
1. Database file `*_db.sql` is deleted after successful Rollback/Recovery.

```
Enabling maintenance mode
You are about to remove current code and/or database tables. Are you sure?[y/N]y
Code rollback is starting ...
Code rollback filename: 1472780740_filesystem_code.tgz
Code rollback file path: /var/www/example.com/magento/2.x/2.x/var/backups/1472780740_filesystem_code.tgz
[SUCCESS]: Code rollback completed successfully.
Media rollback is starting ...
Media rollback filename: 1472780740_filesystem_media.tgz
Media rollback file path: /var/www/example.com/magento/2.x/2.x/var/backups/1472780740_filesystem_media.tgz
[SUCCESS]: Media rollback completed successfully.
DB rollback is starting...
DB rollback filename: 1472780740_db.sql
DB rollback path: /var/www/example.com/magento/2.x/2.x/var/backups/1472780740_db.sql
[SUCCESS]: DB rollback completed successfully.
Please set file permission of bin/magento to executable
Disabling maintenance mode
```

```
www-data@skynet:~/example.com/magento/2.x/2.x$ ls -lht var/backups/
total 43M
-rw-rw-r-- 1 www-data www-data  60K Sep  1 18:31 1472780740_filesystem_media.tgz
-rw-rw-r-- 1 www-data www-data  43M Sep  1 18:31 1472780740_filesystem_code.tgz
```

2.
I do not see any errors showing up in `${webroot}/var/log/system.log`.

**Comment by veloraven** [comment]
@azeemism thank you for your report.
Please, format this issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.

**Comment by azeemism** [comment]
@veloraven the comment has been updated. Thanks.

BTW the numbering reverting back to 1 within each step category is a github issue. It is inconsistent, and the more I try to fix it the worse it gets.

**Comment by BaDos** [comment]
Hi @azeemism 

Thank you for reporting.
We've created internal ticket MAGETWO-58524 to resolve this issue.

**Comment by Tonop** [comment]
2.1.6 the same situation, db backup has been deleted after recovery.

**Comment by magento-engcom-team** [comment]
@azeemism, thank you for your report.
We've created internal ticket(s) MAGETWO-58524 to track progress on the issue.

**Comment by osrecio** [comment]
Working on it

**Comment by osrecio** [comment]
@azeemism This feature will be available in `Magento 2.3`. Now I will work for Backports to have this feature in Magento 2.2 and 2.1.

**Comment by ntuong95** [comment]
Problem still exists in Magento CE 2.2.3 . In fact is has gotten worse. I had too database backups under var/backups. When trying to rollback with the newer backup, system loaded the older db and deleted the new db file. Later when trying to rollback with the older backup ( only db file left ), system deleted this db file and rolled back db to vanilla state.
